src/binary/cursor.rs

```diff
@@ -1063,11 +1105,18 @@ where
     }
 
     fn skip_current_value(&mut self) -> IonResult<()> {
-        if self.cursor.index_at_depth == 0 {
-            Ok(())
-        } else {
-            let bytes_to_skip = self.cursor.value.value_end_exclusive() - self.cursor.bytes_read;
+        let pos = self.cursor.bytes_read;
+        let end = self.cursor.value.value_end_exclusive();
+
+        // Don't skip a value if we haven't called `next()` at the current level
+        if self.cursor.index_at_depth > 0 {
+            // NOPs are not values, so bytes_read can be past the last consumed value (`end`).
+            // This is an artifact of consuming NOP sleds with recursive `next()`, and could be
+            // fixed with refactoring.
+            let bytes_to_skip = end.saturating_sub(pos);
```

Comment:
Two thoughts here:
* I'd prefer to spell `pos` out as `position` if you don't mind
* `saturating_sub` isn't super common, so I added a little explanation
```suggestion
        let position = self.cursor.bytes_read;
        let end = self.cursor.value.value_end_exclusive();

        // Don't skip a value if we haven't called `next()` at the current level
        if self.cursor.index_at_depth > 0 {
            // NOPs are not values, so bytes_read can be past the last consumed value (`end`).
            // This is an artifact of consuming NOP sleds with recursive `next()`, and could be
            // fixed with refactoring. 
            // `saturating_sub` avoids underflow by returning 0 if `position` > `end`.
            let bytes_to_skip = end.saturating_sub(position);
```